### ISSUE TITLE
no-extra-parens should be allowed for destructuring objects in a function parameter

### ISSUE THREAD

**Issue by luislobo** [eslint/eslint#12084]
**What rule do you want to change?**
no-extra-parens

**Does this change cause the rule to produce more or fewer warnings?**
Fewer warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option

**Please provide some example code that this change will affect:**

```js
return SiteService.update(foundSite.id, ({ latitude, longitude } = newSite));
```

**What does the rule currently do for this code?**
   error  Unnecessary parentheses around expression  no-extra-parens

And two related ones: 
    error  'latitude' is not defined                  no-undef
    error  'longitude' is not defined                 no-undef

**What will the rule do after it's changed?**
It should accept parenthesis on parameters that destructure objects

**Are you willing to submit a pull request to implement this change?**
I don't know the project internally fully enough to contribute yet
